static/app/components/keyValueData/card.tsx

```diff
@@ -132,12 +132,42 @@ export function Card({
   );
 }
 
+type ReactFCWithProps = React.FC<CardProps>;
+
+const isReactComponent = (type): type is ReactFCWithProps => {
+  return (
+    typeof type === 'function' ||
+    (typeof type === 'object' && type !== null && 'render' in type)
+  );
+};
+
+// Returns an array of children where null/undefined children and children returning null are filtered out.
+// For example:
+// <Component1/> --> returns a <Card/>
+// {null}
+// <Component2/> --> returns null
+// Gives us back [<Component1/>]
+const filterChildren = (children: ReactNode): ReactNode[] => {
+  return Children.toArray(children).filter((child: React.ReactNode) => {
+    if (isValidElement(child) && isReactComponent(child.type)) {
+      // Render the child and check if it returns null
+      const renderedChild = child.type(child.props);
```

Comment:
This potentially breaks (see issue) when child components use hooks as the hooks are now called in scope of this component.

@Abdkhan14 @AbhiPrasad

Reply:
Yeah good catch - shall we revert this?

Reply:
@Abdkhan14 @AbhiPrasad was tracking down some sentry errors that have been popping up: https://sentry.sentry.io/issues/?project=11276&query=rendered%20more%20hooks&referrer=issue-list&sort=date&statsPeriod=7d

Does seem due to this function, could we find a different solution?